cfme/tests/middleware/test_middleware_server_compliance.py

```diff
@@ -0,0 +1,107 @@
+import fauxfactory
+import pytest
+from cfme.control.explorer.policies import MiddlewareServerCompliancePolicy
+from cfme.middleware.provider.hawkular import HawkularProvider
+from cfme.utils.version import current_version
+from server_methods import (
+    get_eap_server,
+    get_hawkular_server,
+    get_domain_server,
+    verify_server_compliant
+)
+
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.uncollectif(lambda: current_version() < '5.9'),
+    pytest.mark.provider([HawkularProvider], scope="function"),
```

Comment:
Q: are you sure you really need to add a provider for each new test ?

Reply:
Unless it gets removed it won't get re-added